packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -626,10 +625,27 @@ - (void)pause:(FLTTextureMessage*)input error:(FlutterError**)error {
 - (void)setMixWithOthers:(FLTMixWithOthersMessage*)input
                    error:(FlutterError* _Nullable __autoreleasing*)error {
   if ([input.mixWithOthers boolValue]) {
-    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback
-                                     withOptions:AVAudioSessionCategoryOptionMixWithOthers
-                                           error:nil];
+    if ([input.ambient boolValue]) {
+      if (@available(iOS 12.0, *)) {
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient
+                                                mode:AVAudioSessionModeVoicePrompt
+                                             options:AVAudioSessionCategoryOptionMixWithOthers
+                                               error:nil];
+      } else {
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient
+                                         withOptions:AVAudioSessionCategoryOptionMixWithOthers
+                                               error:nil];
+      }
+    } else {
+      [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback
+                                       withOptions:AVAudioSessionCategoryOptionMixWithOthers
+                                             error:nil];
+    }
   } else {
+    if ([input.ambient boolValue]) {
+      [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient error:nil];
+      return;
+    }
     [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback error:nil];
   }
 }
```

Comment:
The iOS changes here will need native unit tests based on a mock `AVAudioSession`, as that's the core of the PR's change.

Reply:
That's right, Thanks. The native test has been added.